docs/7-infrastructure-configuration-management/7.1.3-terraform-modules.md

```diff
@@ -170,7 +170,15 @@ resource "aws_s3_bucket" "s3_bucket" {
   tags = var.tags
 }
 
+resource "aws_s3_bucket_ownership_controls" "s3_bucket" {
+  bucket = aws_s3_bucket.s3_bucket.id
+  rule {
+    object_ownership = "BucketOwnerPreferred"
+  }
+}
+
 resource "aws_s3_bucket_acl" "s3_bucket" {
+  depends_on = [ aws_s3_bucket_ownership_controls.s3_bucket ]
```

Comment:
Without this change `terrafrom validate` would work but trying to `apply` did not, threw this error.

```
 ╷
│ Error: error creating S3 bucket ACL for jburns-dob-s3-override: AccessControlListNotSupported: The bucket does not allow ACLs
│ 	status code: 400, request id: 6805V2BD2BB1444S, host id: 0RCyeU7CQD3A1uG08c3qt+Q/k9xKtI7c5trAo/0Y6Ez8PlIgmRlReElfB1zR2kAhXEEGkvzUMiA9bmEV6cwNqQ==
│ 
│   with module.website_s3_bucket.aws_s3_bucket_acl.s3_bucket,
│   on modules/s3-bucket/main.tf line 6, in resource "aws_s3_bucket_acl" "s3_bucket":
│    6: resource "aws_s3_bucket_acl" "s3_bucket" {
│ 
╵
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.
```

I found the fix by consulting the resource documentation for `aws_s3_bucket` https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_acl#with-private-acl